java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -177,7 +163,7 @@ protected RedisRequest.Builder prepareRedisRequest(Transaction transaction) {
      *     adding a callback id.
      */
     protected RedisRequest.Builder prepareRedisRequest(
-            ClusterTransaction transaction, Optional<Route> route) {
+            BaseTransaction<?> transaction, Optional<Route> route) {
```

Comment:
There was a reason we split this command into Cluster and Standalone transactions.  The standalone doesn't take a route, so there was no reason to create the Optional.empty().  

I would suggest reverting the refactor.  I don't see the benefit of this.

Reply:
Reverted